newIDE/app/src/MainFrame/index.js

```diff
@@ -307,60 +326,53 @@ class MainFrame extends React.Component<Props, State> {
       });
   };
 
-  loadFromSerializedProject = (
+  const loadFromSerializedProject = (
     serializedProject: gdSerializerElement,
-    fileMetadata: ?FileMetadata
-  ): Promise<void> => {
+    fileMetadata: ?FileMetadata,
+    newState: State = state
+  ): Promise<State> => {
     return timePromise(
       () => {
         const newProject = gd.ProjectHelper.createNewGDJSProject();
         newProject.unserializeFrom(serializedProject);
-
-        return this.loadFromProject(newProject, fileMetadata);
+        return loadFromProject(newProject, fileMetadata, newState);
       },
       time => console.info(`Unserialization took ${time} ms`)
     );
   };
 
-  loadFromProject = (
+  const loadFromProject = async (
     project: gdProject,
-    fileMetadata: ?FileMetadata
-  ): Promise<void> => {
-    const { eventsFunctionsExtensionsState } = this.props;
-
-    return this.closeProject().then(() => {
-      // Make sure that the ResourcesLoader cache is emptied, so that
-      // the URL to a resource with a name in the old project is not re-used
-      // for another resource with the same name in the new project.
-      ResourcesLoader.burstAllUrlsCache();
-      // TODO: Pixi cache should also be burst
-
-      return new Promise(resolve => {
-        this.setState(
-          {
-            currentProject: project,
-            currentFileMetadata: fileMetadata,
-          },
-          () => {
-            // Load all the EventsFunctionsExtension when the game is loaded. If they are modified,
-            // their editor will take care of reloading them.
-            eventsFunctionsExtensionsState.loadProjectEventsFunctionsExtensions(
-              project
-            );
-
-            if (fileMetadata) {
-              project.setProjectFile(fileMetadata.fileIdentifier);
-            }
-
-            resolve();
-          }
-        );
-      });
+    fileMetadata: ?FileMetadata,
+    newState: State = state
+  ): Promise<State> => {
+    const { eventsFunctionsExtensionsState } = props;
+
+    newState = await closeProject(newState);
+    // Make sure that the ResourcesLoader cache is emptied, so that
+    // the URL to a resource with a name in the old project is not re-used
+    // for another resource with the same name in the new project.
+    ResourcesLoader.burstAllUrlsCache();
+    // TODO: Pixi cache should also be burst
+    newState = await setState({
+      ...newState,
+      currentProject: project,
+      currentFileMetadata: fileMetadata,
     });
+
```

Comment:
Up to this point, the correct state is "injected" for web-app, but right after this it automatically falls back to its closure, I can't understand how or why this is happening, since it's working fine for electron.